### PR TITLE
QRadar: Bug fix - PAPP-24965: Fix automation trigger issue

### PR DESCRIPTION
**Pull request by ighelani-crest**
Modified Artifact ingestion logic to ingest artifacts in the batch of 1000 artifacts.
Earlier the app was making calls to the save_artifact method for each artifact. and the automation trigger was failing if the last artifact was duplicated.
Logic is modified to collect all the artifacts and then ingest them all at once, by keeping run_automation true for all the artifacts.

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
    - Integration pipeline takes around an hour to complete. And timeout is set to 1 hour.
    - Here is manual pipeline link on latest commit: https://cd.splunkdev.com/phantom-apps/app-tests/-/pipelines/5234811
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [x] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`. No endpoints are exposed
- [x] If this is a new connector or you are adding new actions (No new actions are added)
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [x] Are you introducing any new cryptography modules? If yes, please elaborate their purpose: No
- [x] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [x] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)? No, all sengrep fix are included.
If yes, please provide justification in an additional comment next to the ignored code.

Integration Pipeline:  https://cd.splunkdev.com/phantom-apps/app-tests/-/pipelines/5027576

### CODE REVIEW

**Review by tonyc-phantom — APPROVED** [review]

File: release_notes/unreleased.md [hunk 1]
```diff
@@ -1 +1,3 @@
 **Unreleased**
+* Fixed the automation triggering issue for 'on poll' [PAPP-24628]
+* Fixed a bug for the 'list offenses' action [PAPP-23735]
```
Comment (paulnawat) [inline release_notes/unreleased.md:3]: suggestion: Would it be possible to be more specific about the bug that gets fixed for this action?

File: qradar_connector.py [hunk 2]
```diff
@@ -890,33 +879,51 @@ def _report_back(self, new_offenses, runs):
...
+            'source_data_identifier': offense_id,
+            'cef': offense,
+            'label': 'offense',
+            'run_automation': True
+        }
+
+        # This method will be called in the loop, add offense artifact only if it is not created earlier.
+        # At least one artifact in the list indicates that the offense artifact is already created.
+        if not self._artifacts_list:
+            self._artifacts_list.append(artifact)
```
Comment (paulnawat) [inline qradar_connector.py:903]: What do you think about:

```python
if not self._artifacts_list:
  self.debug_print('_artifacts_list is already populated')
  return

artifact = {
            'name': 'Offense Artifact',
            'container_id': container_id,
            'source_data_identifier': offense_id,
            'cef': offense,
            'label': 'offense',
            'run_automation': True
}
self._artifacts_list.append(artifact)
```

File: qradar_connector.py [hunk 3]
```diff
@@ -890,33 +879,51 @@ def _report_back(self, new_offenses, runs):
...
+
+    def _ingest_collected_artifacts(self, offense_id):
```
Comment (paulnawat) [inline qradar_connector.py:905]: Seeing that this method contains some complicated logic, would it be possible to add some docstring to help us understand this block faster in the future?

**Comment by paulnawat** [comment]
Your description 
```
Modified Artifact ingestion logic to ingest artifacts in the batch of 1000 artifacts.
Earlier the app was making calls to the save_artifact method for each artifact. and the automation trigger was failing if the last artifact was duplicated.
Logic is modified to collect all the artifacts and then ingest them all at once, by keeping run_automation true for all the artifacts.

```
is very clear. Should we include this description in the code as well? maybe on the updated ingestion method

**Review by paulnawat — APPROVED** [review]
lgtm. I left a few suggestions / questions.

File: release_notes/unreleased.md (hunk 1 above)
Comment (ighelani-crest) [reply release_notes/unreleased.md]: Updated the release notes

File: qradar_connector.py (hunk 2 above)
Comment (ighelani-crest) [reply qradar_connector.py]: Included the given suggestion.

File: qradar_connector.py (hunk 3 above)
Comment (ighelani-crest) [reply qradar_connector.py:909]: Added a docstring and comment for artifact creation and ingest function.

**Review by ishans-crest — APPROVED** [review]

**Pull request merged**